x-pack/solutions/security/plugins/security_solution/public/notes/components/delete_confirm_modal.test.tsx

```diff
@@ -0,0 +1,127 @@
+const mockUserClosedDeleteModal = jest.mocked(
+  jest.requireMock('../store/notes.slice').userClosedDeleteModal
+);
+
+describe('DeleteConfirmModal', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders with correct number of notes', () => {
+    const store = createMockStore({
+      ...mockGlobalState,
+      notes: {
+        ...mockGlobalState.notes,
+        pendingDeleteIds: ['1', '2'],
+      },
+    });
+
+    const { getByText } = render(
+      <TestProviders store={store}>
+        <DeleteConfirmModal />
+      </TestProviders>
+    );
+
+    expect(getByText(/Are you sure you want to delete 2 notes/)).toBeInTheDocument();
+  });
+
+  it('dispatches userClosedDeleteModal when cancel is clicked', () => {
+    const store = createMockStore({
+      ...mockGlobalState,
+      notes: {
+        ...mockGlobalState.notes,
+        pendingDeleteIds: ['1'],
+      },
+    });
+
+    const { getByTestId } = render(
+      <TestProviders store={store}>
+        <DeleteConfirmModal />
+      </TestProviders>
+    );
+
+    fireEvent.click(getByTestId('confirmModalCancelButton'));
+    expect(mockUserClosedDeleteModal).toHaveBeenCalled();
+  });
+
+  it('dispatches deleteNotes with correct ids and default refetch when confirm is clicked', () => {
+    const store = createMockStore({
+      ...mockGlobalState,
+      notes: {
+        ...mockGlobalState.notes,
+        pendingDeleteIds: ['1', '2'],
+      },
+    });
+
+    const { getByTestId } = render(
+      <TestProviders store={store}>
+        <DeleteConfirmModal />
+      </TestProviders>
+    );
+
+    fireEvent.click(getByTestId('confirmModalConfirmButton'));
+    expect(mockDeleteNotes).toHaveBeenCalledWith({ ids: ['1', '2'], refetch: true });
+  });
+
+  it('dispatches deleteNotes with refetch: false when specified', () => {
+    const store = createMockStore({
+      ...mockGlobalState,
+      notes: {
+        ...mockGlobalState.notes,
+        pendingDeleteIds: ['1', '2'],
+      },
+    });
+
+    const { getByTestId } = render(
+      <TestProviders store={store}>
+        <DeleteConfirmModal refetch={false} />
+      </TestProviders>
+    );
+
+    fireEvent.click(getByTestId('confirmModalConfirmButton'));
+    expect(mockDeleteNotes).toHaveBeenCalledWith({ ids: ['1', '2'], refetch: false });
+  });
+
+  it('shows loading state when delete is in progress', () => {
+    const store = createMockStore({
+      ...mockGlobalState,
+      notes: {
+        ...mockGlobalState.notes,
+        status: {
+          ...mockGlobalState.notes.status,
+          deleteNotes: ReqStatus.Loading,
+        },
+        pendingDeleteIds: ['1'],
+      },
+    });
+
+    const { getByTestId } = render(
+      <TestProviders store={store}>
+        <DeleteConfirmModal />
+      </TestProviders>
+    );
+
+    const deleteButton = getByTestId('confirmModalConfirmButton');
+    expect(deleteButton.closest('button')).toHaveAttribute('disabled');
```

Comment:
Thanks for adding these tests 🚀 .

You can consider adding tests to `AddNote` component as well which actually takes care of assigning the value of `pendingDeleteIds` so that we are sure that correct list of `notes` are being passed for deletion.